src/main/java/com/exasol/adapter/dialects/SqlGenerationVisitor.java

```diff
@@ -209,22 +246,31 @@ public String visit(final SqlGroupBy groupBy) throws AdapterException {
 
     @Override
     public String visit(final SqlFunctionAggregate function) throws AdapterException {
-        final List<String> argumentsSql = new ArrayList<>();
+        final StringBuilder builder = new StringBuilder();
+        final String functionNameInSourceSystem = this.dialect.getAggregateFunctionAliases()
+                .getOrDefault(function.getFunction(), function.getFunctionName());
+        builder.append(functionNameInSourceSystem);
+        builder.append("(");
+        final List<String> renderedArguments = new ArrayList<>();
         for (final SqlNode node : function.getArguments()) {
-            argumentsSql.add(node.accept(this));
+            renderedArguments.add(node.accept(this));
         }
-        if (function.getFunctionName().equalsIgnoreCase("count") && argumentsSql.isEmpty()) {
-            argumentsSql.add(SqlConstants.ASTERISK);
+        boolean countFunction = function.getFunctionName().equals("COUNT");
+        if (countFunction && renderedArguments.size() > 1) {
+            builder.append("(");
         }
-        String distinctSql = "";
         if (function.hasDistinct()) {
-            distinctSql = "DISTINCT ";
+            builder.append("DISTINCT ");
         }
-        String functionNameInSourceSystem = function.getFunctionName();
-        if (this.dialect.getAggregateFunctionAliases().containsKey(function.getFunction())) {
-            functionNameInSourceSystem = this.dialect.getAggregateFunctionAliases().get(function.getFunction());
+        if (countFunction && renderedArguments.isEmpty()) {
+            renderedArguments.add(SqlConstants.ASTERISK);
         }
-        return functionNameInSourceSystem + "(" + distinctSql + String.join(", ", argumentsSql) + ")";
+        builder.append(String.join(", ", renderedArguments));
+        builder.append(")");
+        if (countFunction && renderedArguments.size() > 1) {
+            builder.append(")");
+        }
```

Comment:
This does not change the result, but I think it makes the source more readable, because the `if` refers to the inner parenthesis:
```suggestion
        if (countFunction && renderedArguments.size() > 1) {
            builder.append(")");
        }
        builder.append(")");
```